File: week-1/binary-gap/binary-gap.py
class Solution:
    def binaryGap(self, N: int) -> int:
        res = 0
        pos = None
        for i in range(0, 32):
            if (N >> i) & 1:
                if pos is not None:
                    res = max(res, i-pos)
                pos = i
                    
        return res

File: week-1/binary-gap/binary-gap.js
/**
 * @param {number} N
 * @return {number}
 */
var binaryGap = function(N) {
    res = 0;
    let pos = null;
    for (let i=0; i<32; i++) {
        if ((N >> i) & 1) {
            if (pos !== null) res = Math.max(res, i-pos);
            pos = i;
        }
    }
    return res;
};
